tests/test-coauthors-plus.php

```diff
@@ -679,6 +699,1434 @@ public function test_update_author_term_when_author_term_not_exist() {
+				$this->author3->user_login,
+				$this->author2->user_login,
+			)
+		);
+
+		$this->assertTrue( $result );
+
+		$second_query = new WP_Query(
+			array(
+				'p' => $post_id,
+			)
+		);
+
+		$this->assertEquals( 1, $second_query->found_posts );
+		$this->assertEquals( $this->author3->ID, $second_query->posts[0]->post_author );
+
+		$post_author_terms = wp_get_post_terms( $post_id, $this->_cap->coauthor_taxonomy );
+
+		$this->assertTrue( is_array( $post_author_terms ) );
+		$this->assertCount( 3, $post_author_terms );
+
+		$author_slugs = array(
+			'cap-' . $this->author2->user_login,
+			'cap-' . $this->author3->user_login,
+			'cap-' . $guest_author_1->user_login,
+		);
+
+		foreach ( $post_author_terms as $term ) {
+			$this->assertInstanceOf( WP_Term::class, $term );
+			$this->assertContains( $term->slug, $author_slugs );
+		}
+
+		$guest_author_term = wp_get_post_terms( $linked_author_1->ID, $this->_cap->coauthor_taxonomy );
+
+		$this->assertTrue( is_array( $guest_author_term ) );
+		$this->assertCount( 1, $guest_author_term );
+		$this->assertEquals( 'cap-' . $linked_author_1->user_login, $guest_author_term[0]->slug );
+	}
+
+	/**
+	 * This test is a combination of a few different aspects that establish some expected plugin behavior.
+	 * Two posts are created to test with. Both have no WP_User as the author (i.e. wp_posts.post_author = 0).
+	 * For the first post:
+	 * 1. A guest author is created, and assigned to the post.
+	 * 2. Since there is no WP_User which is passed, and the GA is NOT being appended, the result should be false.
+	 * 3. The wp_posts.post_author column should still be 0 since this is a GA and not a WP_User or a linked GA.
+	 *
+	 * For the second post:
+	 * 1. A guest author is created, and appended to the post.
+	 * 2. Since we are appending a coauthor, it does not matter if there was already a WP_User author, so result should be true.
+	 * 3. The wp_posts.post_author column should still be 0 since this is a GA and not a WP_User or a linked GA.
+	 *
+	 * Going back to the first post:
+	 * 1. A linked user and a WP_User are created and assigned to this post.
+	 * 2. Result should be true since here we essentially have 2 WP_Users.
+	 * 3. Since there is a WP_User, and it is passed first, the wp_posts.post_author column should be set to the ID for that WP_User.
+	 * 4. There should only be 2 author terms for this post, one for the WP_User, and one for the linked account.
+	 * The term for the GA which was previously assigned is deleted.
+	 *
+	 * Finally, going back to the second post:
+	 * 1. A linked user and a WP_User are created and appended to this post.
+	 * 2. Result should be true since we have 2 WP_Users.
+	 * 3. Here we passed the linked author first, so the wp_posts.post_author column should match the ID for the linked WP_user account.
+	 * 4. There should be 3 author terms for this post, one for the GA, the WP_user, and linked account.
```

Comment:
I greatly appreciate the extra documentation on this test method. But, other than behaviour around when a post starts off with post_author = 0, what else is being tested here that hasn't already been tested above?

Reply:
I deleted this test after realizing, after your comment, that it isn't doing anything that hasn't already been tested with the new tests that were added.